### DISCUSSION TITLE
How to make Popover work inside a scrollable div?

### DISCUSSION THREAD

**Discussion by tuananhlai** [adobe/react-spectrum#5132] [Q&A]
Hi everyone, I have a small question regarding React Spectrum's Popover.

While Popover's position is updated when the body is scrolled, the position doesn't seem to be updated when it is put in a scrollable `div`. The video below demonstrate what I meant. I'm using `ComboBox` element here, which uses Popover under the hood. Here's the [CodeSandBox](https://codesandbox.io/s/bold-hodgkin-g7324l?file=/src/App.js) link.

If anyone can help me resolve this issue, I'd be deeply appreciated. Thank you.

https://github.com/adobe/react-spectrum/assets/42168740/f889a9df-b501-4b78-9b54-94f3f3210501

**Comment by LFDanLu** [comment]
This is actually a bug, the combobox should actually close its popover when the page is scrolled due to issues in updating the popover's placement in cases like the one you've shown above.

**Reply by tuananhlai** [reply]
Is there a known workaround for this? 🤔

**Reply by LFDanLu** [reply]
Short of modifying [this line](https://github.com/adobe/react-spectrum/blob/f29502a37a11c8b294850b6dc3783f43bec85cbd/packages/%40react-aria/overlays/src/usePopover.ts#L92) so that it is `!(isNonModal || !state.isOpen)` via something like [patch-package](https://www.npmjs.com/package/patch-package) (untested btw), no unfortunately.

**Reply by tuananhlai** [reply]
Thank you for your answer. Is this what you meant by modifying the line?

```ts
  let {overlayProps: positionProps, arrowProps, placement} = useOverlayPosition({
    ...otherProps,
    targetRef: triggerRef,
    overlayRef: popoverRef,
    isOpen: state.isOpen,
    onClose: !(isNonModal || !state.isOpen) ? state.close : null
  });
  ```

**Reply by LFDanLu** [reply]
That looks about right

**Reply by tuananhlai** [reply]
Thank you for your time. I've created a bug report issue for this behavior. 🙇

**Comment by tuananhlai** [comment]
Bug report issue created. Closing the discussion as resolved.

https://github.com/adobe/react-spectrum/issues/5139
